File: frontend/src/types/cv/CvConfigurationRequestDto.ts
import { KeyValueObject } from '@/types/KeyValueObject.ts'

export interface CvConfigurationRequestDto {
  includedCvContent: {
    includedWorkExperience?: CvEntrySelectionRequestDto[]
    includedEducation?: CvEntrySelectionRequestDto[]
    includedProjects?: CvEntrySelectionRequestDto[]
    includedSkills?: number[]
  } | undefined

  cvStyle?: string
  cvStyleOptions?: KeyValueObject<string>
}

export interface CvEntrySelectionRequestDto {
  id?: number
  includeDescription?: boolean
}

File: backend/src/main/kotlin/ch/streckeisen/mycv/backend/cv/generator/CvConfigurationRequestDto.kt
package ch.streckeisen.mycv.backend.cv.generator

data class CvConfigurationRequestDto(
    val cvStyle: String?,
    val includedCvContent: IncludedCvContentDto?,
    val cvStyleOptions: Map<String, String>?
)

data class IncludedCvContentDto(
    val includedWorkExperience: List<IncludedCVItem>?,
    val includedEducation: List<IncludedCVItem>?,
    val includedProjects: List<IncludedCVItem>?,
    val includedSkills: List<Long>?,
)

data class IncludedCVItem(
    val id: Long?,
    val includeDescription: Boolean? = true
)